websites/P/Physics Wallah/presence.ts

```diff
@@ -0,0 +1,147 @@
+			presenceData.details = "Studying...";
+			presenceData.state = "My Batches";
+			presenceData.smallImageKey = "studying";
+			presenceData.smallImageText = "Studying";
+		}
+
+		if (Path.includes("batch-overview")) {
+			presenceData.details = "Studying...";
+
+			presenceData.state = `Viewing ${
+				document.querySelector(".bold.text-white").innerHTML
+			}`;
+			presenceData.smallImageKey = "studying";
+			presenceData.smallImageText = "Studying";
+			presenceData.buttons = [
+				{ label: "View Batch", url: document.location.href },
+			];
+		}
+
+		if (Path.includes("batch-video-player")) {
+			const deta = localStorage.getItem("dpp_subject");
+			let detal = ` | ${deta}`;
+
+			if (deta === null) detal = "";
+
+			presenceData.details = `Watching Lecture${detal}`;
+
+			presenceData.state = `${
+				JSON.parse(localStorage.getItem("VIDEO_DETAILS")).topic
+			}`;
+			presenceData.buttons = [
+				{ label: "Watch Lecture", url: document.location.href },
+			];
+
+			updateVideoTimestamps();
+			presenceData.startTimestamp = mediaTimestamps[0];
+			presenceData.endTimestamp = mediaTimestamps[1];
+
+			const video = document.querySelectorAll(".vjs-paused");
+
+			if (video.length > 1) {
+				presenceData.smallImageKey = "paused";
+				presenceData.smallImageText = "Paused";
+			}
+
+			if (video.length === 0) {
+				presenceData.smallImageKey = "watching";
+				presenceData.smallImageText = "Watching a lecture";
+			}
+		}
+	}
+
+	if (Path.startsWith("/watch")) {
+		const deta = localStorage.getItem("dpp_subject");
+		let detal = ` | ${deta}`;
+
+		if (deta === null) detal = "";
+
+		presenceData.details = `Watching Lecture${detal}`;
+
+		presenceData.state = `${
+			JSON.parse(localStorage.getItem("VIDEO_DETAILS")).topic
+		}`;
+		presenceData.smallImageKey = "watching";
+		presenceData.smallImageText = "Watching a lecture";
+		presenceData.buttons = [
+			{ label: "Watch Lecture", url: document.location.href },
+		];
+
+		updateVideoTimestamps();
+		presenceData.startTimestamp = mediaTimestamps[0];
+		presenceData.endTimestamp = mediaTimestamps[1];
+
+		const video = document.querySelectorAll(".vjs-paused");
+
+		if (video.length > 0) {
+			presenceData.smallImageKey = "paused";
+			presenceData.smallImageText = "Paused";
+		}
+
+		if (video.length === 0) {
+			presenceData.smallImageKey = "watching";
+			presenceData.smallImageText = "Watching a lecture";
+		}
+	}
+
+	if (Path.includes("subject-topics")) {
+		const urlParams = new URLSearchParams(window.location.search);
+
+		if (urlParams.has("chapterId") === true) {
+			presenceData.details = urlParams.get("subject");
+			presenceData.state = urlParams.get("topic");
+			presenceData.smallImageKey = "studying";
+			presenceData.smallImageText = "Browsing Resources";
+		} else if (urlParams.has("chapterId") === false) {
+			presenceData.details = urlParams.get("subject");
+			presenceData.state = "Browsing Resources...";
+			presenceData.smallImageKey = "studying";
```

Comment:
ditto

Reply:
> hey man, can I know what is "ditto" supposed to mean?, like i can understand that there are few things which are repeating, but the whole thing?

Ditto as in same as above

Reply:
> > hey man, can I know what is "ditto" supposed to mean?, like i can understand that there are few things which are repeating, but the whole thing?
> 
> Ditto as in same as above, so yes everytime you use that smallimagekey

So, I have to use Imgur links as smallimageKey?

Reply:
> > > hey man, can I know what is "ditto" supposed to mean?, like i can understand that there are few things which are repeating, but the whole thing?
> > 
> > 
> > Ditto as in same as above, so yes everytime you use that smallimagekey
> 
> So, I have to use Imgur links as smallimageKey?

yes preferably in an enum